src/Orleans.Transactions/DistributedTM/TransactionAgent.cs

```diff
@@ -108,14 +68,16 @@ public Task<TransactionalStatus> Commit(ITransactionInfo info)
                 }
             }
 
-            if (writeParticipants == null)
-            {
-                return CommitReadOnlyTransaction(transactionInfo);
-            }
+            TransactionalStatus status = (writeParticipants == null)
```

Comment:
Does there need a try/catch to ensure we track failed txns correctly, or are these async methods 'guaranteed' to not throw?